### DISCUSSION TITLE
How to implement own Certificate

### DISCUSSION THREAD

**Discussion by madmouser1** [mcguirepr89/BirdNET-Pi#537] [Q&A] [answered]
Hi, whom can help me with the correct syntax or an example to utilize a certificate file in caddy rather than relying on Letsencrypt.

Thanks.

**Comment by mcguirepr89** [comment, marked as answer]
Howdy, @madmouser1 -- this is the documentation you need https://caddyserver.com/docs/caddyfile/directives/tls

**Comment by madmouser1** [comment]
Many thanks @mcguirepr89 , it's working now and below is the config changes I made manually to the `/etc/caddy/Caddyfile` file incase it can assist someone else.

**Before:**
```
http:// https://birdnetpi.example.com {
        root * /home/birds/BirdSongs/Extracted
        ...
```
**After:**
```
https://birdnetpi.example.com {
        tls /certs/certificate.crt /certs/key.crt
        root * /home/birds/BirdSongs/Extracted
        ...
```
The Certificates must be in PEM format.

with the logfile stating

```
skipping automatic certificate management because one or more matching certificates are already loaded","domain":"birdnetpi.example.com","server>

enabling automatic HTTP->HTTPS redirects","server_name":"srv0"
```

and so far all is working as expected.

**Reply by mcguirepr89** [reply]
Wonderful --

Hint:

Changing the address line from "`https://birdnetpi.example.com {`" to "`birdnetpi.example.com {`" will enable Caddy's automatic HTTPS redirect, forcing visitors to your site who begin their request over HTTP to be redirected to port 443 for TLS :wink: 

Edit: I didn't notice that your log output already stated HTTP->HTTPS redirect ~~ my B!
